hero_trump/card.py

```diff
@@ -53,27 +59,90 @@ def card_info(self):
               "Tools:", self.tools,
               )
 
-c1 = Card("null", 0, "null", 0, 0, 0, 0, 0, 0, False)
-c2 = Card("null", 1, "null", 0, 0, 0, 0, 0, 0, False)
-c3 = Card("null", 2, "null", 0, 0, 0, 0, 0, 0, False)
-c4 = Card("null", 3, "null", 0, 0, 0, 0, 0, 0, False)
-c5 = Card("null", 4, "null", 0, 0, 0, 0, 0, 0, False)
-c6 = Card("null", 5, "null", 0, 0, 0, 0, 0, 0, False)
-c7 = Card("null", 6, "null", 0, 0, 0, 0, 0, 0, False)
-c8 = Card("null", 7, "null", 0, 0, 0, 0, 0, 0, False)
-c9 = Card("null", 8, "null", 0, 0, 0, 0, 0, 0, False)
-c10 = Card("null", 9, "null", 0, 0, 0, 0, 0, 0, False)
-
-c1.assign_card()
-c2.assign_card()
-c3.assign_card()
-c4.assign_card()
-c5.assign_card()
-c6.assign_card()
-c7.assign_card()
-c8.assign_card()
-c9.assign_card()
-c10.assign_card()
-
-c4.card_info()
+class Deck:
+    def __init__(
+        self,
+        cards,
+    ):
+        self.cards = cards
+        self.length = len(cards)
+
+    def read_cards(self):
+        with open('Cards.csv', 'r') as file:  
+            reader = csv.reader(file)
+            for row in reader:
+                newcard = Card()
+                newcard.assign_card(row)
+                cards.append(newcard)
+
+    def add_card(new_card):
+        i = new_card
+        if i in cards:
+            print("This card is already in the deck.")
+        else:
+            cards.append(new_card)
+
+    def remove_card(card_to_remove):
+        i = card_to_remove
+        if i in cards:
+            cards.remove(card_to_remove)
+        else:
+            print("This card is already not in the deck.")
+
+    def find_card(card_to_find):
+        i = card_to_find
+        if i in cards:
+            print(card_to_find)
+        else:
+            print(card_to_find, "cannot be found in the deck.")
+
+    def shuffle_cards():
+        random.shuffle(cards)
+
+
+cards = []
+
+mydeck = Deck(cards)
+
+mydeck.read_cards()
+
+print(cards)
+
+Deck.find_card(cards[2])     #test for finding a card in the deck
+
+Deck.shuffle_cards()    #test for shuffling cards
+
+print(cards[2].category)
+
+Deck.remove_card(cards[5])   #test for removing a card from the deck
+
+print(cards)
+
+
+def create_app():
+    app = FastAPI(
+        title="Server",
+        description="database for cards",
+        version="0.0.1"
+    )
+    return app
+
+app = create_app()
+
+@app.get("/hello")
+async def main():
+    return("Welcome!")
+
+@app.get("/card/")
+async def show_cards(card: Card):
+    return card
+
+@app.post("/cards/")
+async def create_card(card: Card):
+   Deck.add_card(card)
```

Comment:
Deck is the class, you need to have an instance of that class, like my_deck or your_deck or player1_deck, player2_deck. then you can add a card.

Reply:
I suspect this will also struggle to create the card object because you haven't created a data model for the card. Therefore when the api gets a post request (which is basically a string or a dict), it won't know how to convert that into a card object. It may seem obvious to you that "name" in the dictionary needs to be the "name" of the card, but you have to tell fast api. Look at the data model they create for "item" in the tutorial here: https://fastapi.tiangolo.com/tutorial/body/